unittests/runtime-io/io.cpp

```diff
@@ -470,6 +473,11 @@ BOOST_AUTO_TEST_CASE(time) {
   BOOST_CHECK(tt >= 1573756117);
 }
 
+BOOST_AUTO_TEST_CASE(kompiledDir) {
+  auto dir = hook_KREFLECTION_kompiledDir();
+  BOOST_CHECK_EQUAL(0, strcmp(dir->data, kompiled_directory));
```

Comment:
Let's use memcmp here because dir->data isn't necessarily null terminated

Reply:
The LLVM code that constructs the symbol (see [docs here](https://llvm.org/doxygen/classllvm_1_1ConstantDataArray.html#a3edef3fa47c611d3d10606591213e57b) guarantees that it will be a null-terminated C string.

If we don't want to rely on that behaviour, I will need to also go back and bake in a `kompiled_directory_length` global so that we know how many bytes to `memcmp` against.